### PR TITLE
[Task #3] Replace domain models with DTO in data services #3

### LINKED ISSUE #6: Implement DTO(records) + Mapster
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/records 

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record

Mapster (must to use ProjectToType)
https://github.com/MapsterMapper/Mapster/wiki/Basic-usages

### PR DESCRIPTION
**Pull request by Aleksander-Drumev-WA**
#3

### CODE REVIEW

File: WA.Pizza.Infrastructure/DTO/Catalog/CatalogItemToBasketItemRequest.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace WA.Pizza.Infrastructure.DTO.Catalog
+{
+    public record CatalogItemToBasketItemRequest
+    {
+		public int? BasketId { get; init; }
+
+		public int CatalogItemId { get; init; }
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/DTO/Catalog/CatalogItemToBasketItemRequest.cs]: spaces

File: WA.Pizza.Infrastructure/DTO/Orders/OrderDTO.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using WA.Pizza.Core.Models;
+
+namespace WA.Pizza.Infrastructure.DTO.Orders
+{
+    public record OrderDTO
+    {
+		public int Id { get; init; }
+
+		public decimal Total { get; init; }
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/DTO/Orders/OrderDTO.cs]: Spaces

File: WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs [hunk 3]
```diff
@@ -18,34 +18,29 @@ public BasketDataService(AppDbContext dbContext)
 
 
 
-        public ICollection<GetBasketDTO> GetBasketItems(int basketId)
+        public ICollection<BasketDTO> GetBasketItems(int basketId)
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs]: Method tell me that I can get all basket items by basket id but...

File: WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs [hunk 4]
```diff
@@ -56,16 +51,13 @@ public async Task<int> AddItemToBasketAsync(int? userId, int? basketId, CatalogI
         // I think user can only change the quantity
         public async Task<int> UpdateItemAsync(UpdateBasketItemDTO updatedBasketItem)
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs]: UpdateBasketItemDTO?

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs [hunk 5]
```diff
@@ -36,10 +36,20 @@ public ICollection<ListCatalogItemsDTO> GetAllAsync()
         {
             var catalogItems = _dbContext
                 .CatalogItems;
-            
+
             return catalogItems.ProjectToType<ListCatalogItemsDTO>().ToList();
         }
 
+        public async Task<CatalogItemDTO> GetOneAsync(int catalogItemId)
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs]: GetOne what?

File: WA.Pizza.Infrastructure/DTO/Catalog/CatalogItemToBasketItemRequest.cs (hunk 1 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/DTO/Catalog/CatalogItemToBasketItemRequest.cs]: Removed.

File: WA.Pizza.Infrastructure/DTO/Orders/OrderDTO.cs (hunk 2 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/DTO/Orders/OrderDTO.cs]: Removed.

File: WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs (hunk 3 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs]: So just GetBasket (method name)?

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs (hunk 5 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs]: I Changed it.

**Review by var1ap-WA — APPROVED** [review]

**Pull request merged**